OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -162,8 +162,8 @@ public class BuildingCategories
 		[Desc("Terrain types which are considered water for base building purposes.")]
 		public readonly HashSet<string> WaterTerrainTypes = new HashSet<string> { "Water" };
 
-		[Desc("Avoid enemy actors nearby when searching for a new resource patch. Should be somewhere near the max weapon range.")]
-		public readonly WDist HarvesterEnemyAvoidanceRadius = WDist.FromCells(8);
+		[Desc("Which harvester handling module to use. Leave empty to disable.")]
+		public readonly string HarvesterModule = "default-harvester-module";
```

Comment:
Having specifically behaving modules referred to via strings like this sets off warning bells in my head.  What if someone references a support power module type here, or something different?

If we want to refer to specific types of module here, then we should come up with a way to annotate them - perhaps using a new attribute with a tag type.

A much better fix would be to not have specific `*Modules` at all, and instead a generic `Modules` list, with `HackyAI` itself not caring about what each module is/does.